sale_product_seasonality/models/sale_order.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2021 Camptocamp SA
+# @author: Julien Coux <julien.coux@camptocamp.com>
+# @author: Simone Orsi <simone.orsi@camptocamp.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+
+from ..utils import roundTime
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    # TODO: any better name for this field?
+    date_end = fields.Datetime()
+    seasonal_config_id = fields.Many2one(
+        string="Seasonal configuration",
+        comodel_name="seasonal.config",
+        default=lambda self: self._default_seasonal_config_id(),
+    )
+    season_allowed_product_ids = fields.Many2many(
+        string="Season allowed products",
+        comodel_name="product.product",
+        column1="sale_id",
+        column2="product_id",
+        compute="_compute_season_allowed_product_ids",
+    )
+
+    def _default_seasonal_config_id(self):
+        return self.env.user.company_id.default_seasonal_config_id
+
+    def _get_allowed_products(self, date):
+        self.ensure_one()
+        products = self.env["product.product"]
+        if date:
+            config_lines = self.mapped("seasonal_config_id.line_ids")
+            products = config_lines.filtered(lambda l: l.is_sale_ok(date)).mapped(
+                "product_id"
+            )
+        return products
+
+    @api.depends("commitment_date", "seasonal_config_id.line_ids")
+    def _compute_season_allowed_product_ids(self):
+        for sale in self:
+            value = [(5, 0)]
+            if sale.commitment_date:
+                products = sale._get_allowed_products(sale.commitment_date)
+                value = [(6, 0, products.ids)]
+            sale.season_allowed_product_ids = value
+
+    def _round_dates(self):
+        """TODO"""
+        commitment_date = roundTime(dt=self.commitment_date, minutes=5)
```

Comment:
Maybe we want to have the number of minutes in settings to be able to round for 5,10,30 or whatever?

Reply:
good point, I add a TODO for the next round, not doing  it now